src/Kanvas/Users/Actions/SwitchCompanyBranchAction.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Kanvas\Users\Actions;
+
+use Kanvas\Users\Models\Users;
+use Kanvas\Companies\Models\CompaniesBranches;
+use Kanvas\Companies\Models\Companies;
+use Kanvas\Companies\Branches\Repositories\CompaniesBranches as CompaniesBranchesRepository;
+
+class SwitchCompanyBranchAction
+{
+    /**
+     * __construct
+     * @param Users $user
+     * @param int $companyId
+     * @return void
+     */
+    public function __construct(
+        protected Users $user,
+        protected int $companyBranchId
+    ) {
+    }
+
+    /**
+     * execute
+     *
+     * @return Users
+     */
+    public function execute(): Users
+    {
+        if ($branch = CompaniesBranchesRepository::getById((int)$this->companyBranchId, $this->user->id)) {
```

Comment:
@FredPeal this is not needed use UserRepository::belongsToCompanyBranch , for this specific function